### PR TITLE
fix: MavenBinaryInstaller Setting loose POSIX file permissions is sec…

### PR DESCRIPTION
**Pull request by kevinhagel**
…urity-sensitive java:S2612 #532

- added commons-lang3 dependency to webforj-install-bbj-plugin pom.xml
- using SystemUtils to test for MAC or Linux systems, avoiding setting posix permissions on windows.
- Using PosixFilePermission constants to set the permissions for mvn and mvn.cmd to rwxr-xr-x
- clean up imports.

### CODE REVIEW

File: webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java [hunk 1]
```diff
@@ -82,14 +85,31 @@ static void installMaven(String directory) throws IOException {
...
+      perms.add(PosixFilePermission.OTHERS_READ); // Non-Compliant
+      perms.remove(PosixFilePermission.OTHERS_WRITE); // Compliant
+      perms.add(PosixFilePermission.OTHERS_EXECUTE); // Compliant
 
-    Files.setPosixFilePermissions(Path.of(directory + "/" + BINDIR + "/bin/mvn"),
-        PosixFilePermissions.fromString("rwxrwxr-x"));
+      Path binDirectory = Path.of(directory, BINDIR, "bin");
 
-    Files.delete(Path.of(fileZip));
+      Files.setPosixFilePermissions(binDirectory.resolve("mvn.cmd"), perms);
```
Comment (hyyan) [inline webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java:106]: If this is intended for Mac and Linux, then why is there a need to modify mvn.cmd?

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java:106]: good question.  Why was he doing it in the first place?
In this project, stuff like this happens, i always get lots of noise about changing this or that.  I left it here so Stephan could join the conversation and tell us why he never tested for Windows even though his comment said it was.

Comment (hyyan) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java:106]: @kevinhagel Did you test this on windows yourself ?

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: no, I have not.

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: I guess it ends here.  I can't test it on windows. I have no way of doing that.

Comment (hyyan) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: I still need to perform local testing, after which I will guide Matthew to conduct tests on his Windows machine.

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: Ok, I just got it to finally run.  A licensing problem.
And i installed the fat jar, there are no other jars in the plugins/DWCJ/lib directory
I checked out webforj-hello-world and ran mvn clean install on it.  I can attach the log, you can see the maven beinary installer running successfully.

[webforj-install-bbj-plugin.log](https://github.com/webforj/webforj/files/14603337/webforj-install-bbj-plugin.log)

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: stop ... something wrong with the bbj subdirecotry.

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java]: Ok, I found it.  Our old friend Stupid Windows back slash file path separator.  The fix for the zip slip problem requires path checking, I did not see that some of the paths arrived in windows form, some in unix form, and since the test was String ...  I'll get it in today. I need to clean up other stuff first

File: webforj-plugins/webforj-install-bbj-plugin/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
     >
       <PatternLayout pattern="${LOG_PATTERN}"/>
       <Policies>
-        <SizeBasedTriggeringPolicy size="500KB"/>
+        <SizeBasedTriggeringPolicy size="1024KB"/>
```
Comment (hyyan) [inline webforj-plugins/webforj-install-bbj-plugin/src/main/resources/log4j2.xml:23]: Why this change ?

**Comment by hyyan** [comment]
@StephanWald 

We will proceed with the proposed change for now. We can always revisit it if any corner cases arise. Closing this issue is important. With the proposed fix, the install plugin works on both Windows and Linux-based OS. Kevin tested it on both Mac and Windows, I tested again on Mac, and Matthew tested on Windows. It looks good to go.

**Comment by kevinhagel** [comment]
Is this ready for merge?

File: webforj-plugins/webforj-install-bbj-plugin/src/main/resources/log4j2.xml (hunk 2 above)
Comment (hyyan) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/resources/log4j2.xml:23]: ???

Comment (kevinhagel) [reply webforj-plugins/webforj-install-bbj-plugin/src/main/resources/log4j2.xml]: done.  It is a preference, I rolled it back, will make argument for it again in the future
